src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointMessageWriter.cs

```diff
@@ -18,15 +18,33 @@ namespace Microsoft.Azure.SignalR
     internal class MultiEndpointMessageWriter : IServiceConnectionContainer
     {
         private readonly ILogger _logger;
-        internal IReadOnlyCollection<HubServiceEndpoint> TargetEndpoints { get; }
+
+        internal HubServiceEndpoint[] TargetEndpoints { get; }
```

Comment:
Can this be private readonly?

Reply:
Not for now, there is a test `CreateServiceHubContext_WithReferenceNotEqualEndpoints` depends on it.